Scripts/script-RegexGroups.yml

```
@@ -0,0 +1,56 @@
+commonfields:
+  id: RegexGroups
+  version: -1
+name: RegexGroups
+script: |
+  match_target = demisto.args()['value']
+  regex_pattern = re.compile(r'{}'.format(demisto.args()['regex']))
+  capture_groups = demisto.args().get('groups')
+  dict_keys = demisto.args().get('keys')
+
+  if capture_groups:
+      capture_groups = capture_groups.split(',')
+
+  if dict_keys:
+      dict_keys = dict_keys.split(',')
+
+  pattern_match = re.match(regex_pattern, match_target)
+  matches = []
+  if pattern_match:
+      for i in pattern_match.groups():
+          matches.append(i)
+
+  if capture_groups:
+      matches = [matches[int(x)] for x in capture_groups]
+
+  if dict_keys:
+      if len(dict_keys) != len(matches):
+          matches = 'Error: Number of keys does not match number of items'
```

better to return an error entry